### PR TITLE
ci(helm): add `--dev` when doing pull_request too

### PR DESCRIPTION
**Pull request by lahabana**
Signed-off-by: Charly Molter <charly.molter@konghq.com>

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by michaelbeaumont — DISMISSED** [review]

**Review by michaelbeaumont — APPROVED** [review]

File: .github/workflows/helm-release.yaml [hunk 1]
```diff
@@ -76,36 +77,16 @@ jobs:
...
-          curl -L "https://github.com/helm/chart-releaser/releases/download/v${CR_VERSION}/chart-releaser_${CR_VERSION}_linux_amd64.tar.gz"
-          | sudo tar xvz --directory /usr/bin cr
-      - name: Download packaged chart
-        uses: actions/download-artifact@v3
-        with:
-          name: ${{ needs.package.outputs.filename }}
-          path: .cr-release-packages
       - name: Generate GitHub app token
         id: github-app-token
+        if: github.event.inputs.release == 'true'
```
Comment (michaelbeaumont) [inline .github/workflows/helm-release.yaml:82]: Hmm I kind of liked the job level separation and just having one top level place where things are skipped, seems less error prone

File: tools/releases/helm.sh [hunk 2]
```diff
@@ -75,6 +75,13 @@ function package {
 }
 
 function release {
+  if [ "${dev}" == "true" ]; then
```
Comment (michaelbeaumont) [inline tools/releases/helm.sh:78]: Hmm shouldn't we just not run this if it's not supposed to run?

File: .github/workflows/helm-release.yaml (hunk 1 above)
Comment (lahabana) [reply .github/workflows/helm-release.yaml:82]: I don't mind either way. It was just a lot of duplication (especially once you add the cache as well)

File: tools/releases/helm.sh (hunk 2 above)
Comment (lahabana) [reply tools/releases/helm.sh:78]: I don't mind (in practice it doesn't run as it's gated in the action).

Comment (michaelbeaumont) [reply tools/releases/helm.sh:78]: then IMO it's just noise

**Review by michaelbeaumont — APPROVED** [review]

**Pull request merged**